### PR TITLE
Edit deck

### PR DESCRIPTION
**Pull request by malisara**

### CODE REVIEW

File: word_nerd/word_nerd/urls.py [hunk 1]
```diff
@@ -35,4 +35,6 @@
     path('new-deck/', deck_views.NewDeckAPIView.as_view(), name='new_deck'),
     path('my-languages/<int:language_pk>/decks',
          deck_views.GetUserDecksAPIView.as_view(), name='my_decks'),
+    path('deck/<int:pk>/edit', deck_views.EditDeckAPIView.as_view(),
```
Comment (inejc) [inline word_nerd/word_nerd/urls.py]: Maybe use `decks/...` for consistency

File: word_nerd/decks/views.py [hunk 2]
```diff
@@ -36,3 +36,22 @@ def get(self, request, language_pk):
...
+        try:
+            deck = Deck.objects.get(id=pk, owner=request.user)
+        except ObjectDoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        serializer = DeckSerializer(deck, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response({'detail': 'Deck is updated.',
+                             'data': serializer.data})
```
Comment (inejc) [inline word_nerd/decks/views.py]: Why are you sending `data` back? It's not needed IMHO

File: word_nerd/decks/tests/test_update_deck.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        self.assertEqual(deck.name, 'newname')
+        self.assertTrue(deck.public)
+        self.assertEqual(response.data['detail'], 'Deck is updated.')
+        self.assertEqual(response.data['data'],
+                         {'name': 'newname', 'public': True})
+
+
+def _autogenerated_deck(client):
+    # New deck is automatically created after user adds a new language
+    # Deck attributes: deck.name = 'Other cards', deck.public = False,
```
Comment (inejc) [inline word_nerd/decks/tests/test_update_deck.py]: NIT: remove the trailing `,` 😛

File: word_nerd/decks/views.py [hunk 4]
```diff
@@ -36,3 +36,22 @@ def get(self, request, language_pk):
...
+
+
+class EditDeckAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, pk):
+        try:
+            deck = Deck.objects.get(id=pk, owner=request.user)
+        except ObjectDoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (inejc) [inline word_nerd/decks/views.py:48]: Would it be better to return `HTTP_400_BAD_REQUEST` here as well? Not sure

File: word_nerd/decks/tests/test_update_deck.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        self.assertEqual(Deck.objects.count(), 1)
+        deck = Deck.objects.get(id=1)
+        self.assertEqual(deck.name, 'newname')
+        self.assertTrue(deck.public)
+        self.assertEqual(response.data['detail'], 'Deck is updated.')
+        self.assertEqual(response.data['data'],
+                         {'name': 'newname', 'public': True})
+
+
+def _autogenerated_deck(client):
```
Comment (inejc) [inline word_nerd/decks/tests/test_update_deck.py]: Maybe a better name would be `_register_user_with_a_language`?

File: word_nerd/decks/views.py (hunk 4 above)
Comment (malisara) [reply word_nerd/decks/views.py:48]: I feel like it makes more sense to return HTTP_404_NOT_FOUND when the deck/language instance with a specific PK doesn't exist and HTTP_400_BAD_REQUEST when a serializer is invalid :smiley:

**Review by inejc — APPROVED** [review]

**Pull request merged**